docs/design/yarn-v1.md

```diff
@@ -0,0 +1,169 @@
+# Cachi2 Yarn v1 (Yarn Classic) Prefetch Design
+
+## Yarn Overview
+[Yarn](https://github.com/yarnpkg/yarn) is a package manager for JavaScript. While Yarn v1.x is no longer actively developed and has been
+succeeded by [Yarn v2+ (Berry)](https://github.com/yarnpkg/berry), it [remains widely used](https://npmtrends.com/yarn).
+
+## Cachito Implementation Overview
+Cachito supports the Yarn 1.x package manager. It directly parses package.json files and uses our [pyarn](https://github.com/containerbuildsystem/pyarn)
+library to parse yarn.lock files. Dependencies are downloaded from their resolved URLs in the yarn.lock file without executing yarn itself.
+
+Registry dependencies are fetched through a proxy repository in Nexus and the user's build is configured to use that proxy as the registry.
+
+Non-registry dependencies are packed and uploaded to a hosted Nexus repository. The versions of these non-registry dependencies are
+updated in package.json and yarn.lock to match those in the hosted repository.
+
+## Proposed Cachi2 Implementation Overview
+
+### Specifying a Yarn Package in a Cachi2 Request
+Cachi2 users will specify ```yarn``` as the package manager or package type for all versions of Yarn. The presence of the ```__metadata``` field in yarn.lock
```

Comment:
Are we intending to parse out the __metadata field from the lockfile with a regex or are we taking a more conservative route: try parsing with V2+ and if it fails then try V1? I'm asking just out of curiousity, it's an implementation detail.

Reply:
Correction, `__metadata` is already the parsed out representation, sorry for the confusion, looking at https://github.com/yarnpkg/berry/blob/13d5b3041794c33171808fdce635461ff4ab5c4e/packages/yarnpkg-core/sources/Project.ts#L434 they just search for a static string sequence, shall we do the same to determine?

Reply:
I had thought to do pretty much exactly what they do:

1. Check for the `__metadata` field, if present assume v2+ lockfile
2. Check for `yarn lockfile v1`, if present assume v1 lockfile ([we also do this in pyarn](https://github.com/containerbuildsystem/pyarn/blob/c947823027b421c7ec42e247246d9986fbc2843d/pyarn/lockfile.py#L32))
3. Error if neither condition satisfied, since we cannot determine the lockfile version